lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -252,80 +291,153 @@ def _transform_state(server):
-                continue
-
-        self.info = self._transform_state(self._post('servers', data))
-        self.wait_for_state(('running', ))
-
-    def delete_server(self):
-        self._delete('servers/%s' % self.info['uuid'])
-        self.wait_for_state(('absent', ))
-
-    def start_server(self):
-        self._post('servers/%s/start' % self.info['uuid'])
-        self.wait_for_state(('running', ))
+        # Timeout succeeded
+        if server_info.get('name') is not None:
+            msg = "Timeout while waiting for a state change on server %s to states %s. " \
+                  "Current state is %s." % (server_info.get('name'), states, server_info.get('state'))
+        else:
+            name_uuid = self._module.params.get('name') or self._module.params.get('uuid')
+            msg = 'Timeout while waiting to find the server %s' % name_uuid
+
+        self._module.fail_json(msg=msg)
+
+    def _start_stop_server(self, server_info, target_state="running", ignore_diff=False):
+        actions = {
+            'stopped': 'stop',
+            'running': 'start',
+        }
+
+        server_state = server_info.get('state')
+        if server_state != target_state:
+            self._result['changed'] = True
+
+            if not ignore_diff:
+                self._result['diff']['before'].update({
+                    'state': server_info.get('state'),
+                })
+                self._result['diff']['after'].update({
+                    'state': target_state,
+                })
+            if not self._module.check_mode:
+                self._post('servers/%s/%s' % (server_info['uuid'], actions[target_state]))
+                server_info = self._wait_for_state((target_state, ))
+
+        return server_info
+
+    def _update_param(self, param_key, server_info, requires_stop=False):
+        param_value = self._module.params.get(param_key)
+        if param_value is None:
+            return server_info
+
+        if 'slug' in server_info[param_key]:
+            server_v = server_info[param_key]['slug']
+        else:
+            server_v = server_info[param_key]
+
+        if server_v != param_value:
+            # Set the diff output
+            self._result['diff']['before'].update({param_key: server_v})
+            self._result['diff']['after'].update({param_key: param_value})
+
+            if server_info.get('state') == "running":
+                if requires_stop and not self._module.params.get('force'):
+                    self._module.warn("Some changes won't be applied to running servers. "
+                                      "Use force=yes to allow the server '%s' to be stopped/started." % server_info['name'])
+                    return server_info
+
+            # Either the server is stopped or change is forced
+            self._result['changed'] = True
+            if not self._module.check_mode:
+                # Remember the state of the server before we ensure it is stopped
```

Comment:
With the changed code this comment is now wrong. Please move to the place where the server state is actually saved.